sql/core/src/test/resources/sql-tests/inputs/udf/pgSQL/udf-join.sql

```diff
@@ -0,0 +1,2187 @@
+---select udf(c.name), udf(ss.code), udf(ss.b_cnt), udf(ss.const)
+---from c left join
+---  (select a.code, coalesce(b_grp.cnt, 0) as b_cnt, -1 as const
+---   from a left join
+---     (select udf(count(1)) as cnt, udf(b.a) from b group by b.a) as b_grp
+---     on udf(a.code) = udf(b_grp.a)
+---  ) as ss
+---  on (udf(c.a) = udf(ss.code))
+---order by c.name;
+select udf(c.name), udf(ss.code), udf(ss.b_cnt), udf(ss.const)
+from c left join
+  (select a.code, coalesce(b_grp.cnt, 0) as b_cnt, -1 as const
+   from a left join
+     (select udf(count(1)) as cnt, b.a from b group by b.a) as b_grp
+     on udf(a.code) = udf(b_grp.a)
+  ) as ss
+  on (udf(c.a) = udf(ss.code))
+order by c.name;
+
+-- rollback;
+
+--
+-- test incorrect handling of placeholders that only appear in targetlists,
+-- per bug #6154
+--
+---comment out for now  query 137
+---promblem caused by SELECT udf(sub5.key5),
+--+org.apache.spark.sql.AnalysisException
+---+cannot resolve '`sub4.key5`' given input columns: [sub4.CAST(udf(cast(key5 as string)) AS INT), sub3.key3, sub4.value2]; line 13 pos 11
+---SELECT * FROM
+---( SELECT 1 as key1 ) sub1
+---LEFT JOIN
+---( SELECT sub3.key3, sub4.value2, COALESCE(sub4.value2, 66) as value3 FROM
+---    ( SELECT 1 as key3 ) sub3
+---    LEFT JOIN
+---    ( SELECT udf(sub5.key5), udf(COALESCE(sub6.value1, 1)) as value2 FROM
+---        ( SELECT 1 as key5 ) sub5
+---        LEFT JOIN
+---        ( SELECT 2 as key6, 42 as value1 ) sub6
+---        ON udf(sub5.key5) = udf(sub6.key6)
+---    ) sub4
+---    ON udf(sub4.key5) = udf(sub3.key3)
+---) sub2
+---ON udf(sub1.key1) = udf(sub2.key3);
+SELECT * FROM
+( SELECT 1 as key1 ) sub1
+LEFT JOIN
+( SELECT sub3.key3, sub4.value2, COALESCE(sub4.value2, 66) as value3 FROM
+    ( SELECT 1 as key3 ) sub3
+    LEFT JOIN
+    ( SELECT sub5.key5, udf(COALESCE(sub6.value1, 1)) as value2 FROM
+        ( SELECT 1 as key5 ) sub5
+        LEFT JOIN
+        ( SELECT 2 as key6, 42 as value1 ) sub6
+        ON udf(sub5.key5) = udf(sub6.key6)
+    ) sub4
+    ON udf(sub4.key5) = udf(sub3.key3)
+) sub2
+ON udf(sub1.key1) = udf(sub2.key3);
+
+-- test the path using join aliases, too
+---comment out for now query 138
+---problem caused by  udf(sub3.key3)
+---+org.apache.spark.sql.AnalysisException
+---+cannot resolve '`sub2.key3`' given input columns: [sub2.CAST(udf(cast(key3 as string)) AS INT), sub2.CAST(udf(cast(value2 as string)) AS INT), sub1.key1, sub2.value3]; line 15 pos 15
+---SELECT * FROM
+---( SELECT 1 as key1 ) sub1
+---LEFT JOIN
+---( SELECT udf(sub3.key3), udf(value2), udf(COALESCE(value2, 66)) as value3 FROM
+---    ( SELECT 1 as key3 ) sub3
+---    LEFT JOIN
+---    ( SELECT sub5.key5, COALESCE(sub6.value1, 1) as value2 FROM
+---        ( SELECT 1 as key5 ) sub5
+---        LEFT JOIN
+---        ( SELECT 2 as key6, 42 as value1 ) sub6
+---        ON sub5.key5 = sub6.key6
+---    ) sub4
+---    ON sub4.key5 = sub3.key3
+---) sub2
+---ON sub1.key1 = sub2.key3;
```

Comment:
I think here is the problem? Because there is no key3 in sub2 now. You might need to write:

```
...
SELECT udf(sub3.key3) as key3, udf(value2), udf(COALESCE(value2, 66)) as value3 FROM
...
```